plugin/src/tryGetObjects.lua

```diff
@@ -0,0 +1,88 @@
+local Fmt = require(script.Parent.Parent.Fmt)
+local Log = require(script.Parent.Parent.Log)
+local PatchSet = require(script.Parent.PatchSet)
+local Promise = require(script.Parent.Parent.Promise)
+
+local function tryGetObjects(instanceMap, apiContext, patch)
+	assert(PatchSet.validate(patch))
+
+	local unappliedPatch = PatchSet.newEmpty()
+
+	-- GetObjects won't help with anything that failed to remove
+	unappliedPatch.removed = patch.removed
+	-- TODO: Implement this? Do we have to?
+	unappliedPatch.added = patch.added
+
+	local assetsToRequest = {}
+	local receiveCallbacks = {}
+
+	Log.trace("tryGetObjects({:#?})", patch)
+
+	-- TODO: added?
+
+	-- GetObjects only create instances, we can't update the properties of existing ones.
+	-- Instead, just create them again, move their children, and replace the instance.
+	for _, update in ipairs(patch.updated) do
+		table.insert(assetsToRequest, update.id)
+		table.insert(unappliedPatch.updated, update)
+
+		receiveCallbacks[update.id] = function(newInstance)
+			table.remove(unappliedPatch.updated, table.find(unappliedPatch.updated, update))
+
+			local oldInstance = instanceMap.fromIds[update.id]
+
+			-- TODO: What if oldInstance is nil?
+			-- TODO: This will dupe children if the rbxm is not empty.
```

Comment:
I don't think this TODO is actually accurate, the assets being created don't have any children. The bug is from something else 🤷